src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""YOLOv9 model implementations."""
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING, Literal
+
+from otx.algo.detection.backbones import GELAN
+from otx.algo.detection.detectors import SingleStageDetector
+from otx.algo.detection.heads import YOLOHead
+from otx.algo.detection.losses.yolov9_loss import BCELoss, BoxLoss, DFLoss, YOLOv9Criterion
+from otx.algo.detection.necks import YOLONeck
+from otx.algo.detection.utils.yolov7_v9_utils import Vec2Box
+from otx.core.config.data import TileConfig
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.metrics.fmeasure import MeanAveragePrecisionFMeasureCallable
+from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
+from otx.core.model.detection import OTXDetectionModel
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+    from typing_extensions import Self
+
+    from otx.core.metrics import MetricCallable
+    from otx.core.schedulers import LRSchedulerListCallable
+    from otx.core.types.label import LabelInfoTypes
+
+
+PRETRAINED_WEIGHTS: dict[str, str] = {
+    "yolov9_s": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-s.pt",
+    "yolov9_m": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-m.pt",
+    "yolov9_c": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-c.pt",
+}
+
+
+def _load_from_state_dict_for_yolov9(
```

Comment:
I am not sure whether it is proper place for that... 
Maybe it is better to create separate class like YOLOSingleStageDetector, inherit from SingleStageDetector and reload _load_from_state_dict or add it somehow to the SingleStageDetector.  What do you think?

Reply:
Thanks for the suggestion, done at e9b95ac.